### ISSUE TITLE
Is there a way to allow an output cell to scroll by its self? 

### ISSUE THREAD

**Issue by ArEnSc** [jupyter/notebook#1090]

**Comment by willingc** [comment]
@ArEnSc If you ran this code in Python 3 with a Jupyter notebook, you would be able to scroll through the output:

<img width="1137" alt="screen shot 2016-02-14 at 5 32 34 pm" src="https://cloud.githubusercontent.com/assets/2680980/13037980/02ac0bc2-d341-11e5-9de7-a21238cf8bee.png">

**Comment by ArEnSc** [comment]
I would like it to scroll by it self to the latest printed value is that possible ?

**Comment by takluyver** [comment]
I don't think we have an option to do that. You can probably do it with a [notebook extension](http://jupyter-notebook.readthedocs.org/en/latest/examples/Notebook/rstversions/JavaScript%20Notebook%20Extensions.html), but I don't know quite what JS you'd need.

**Comment by JamiesHQ** [comment]
@ArEnSc : We're doing a little housekeeping on our issue log and noticed this thread from 2016. Were you able to find a solution to your issue? Please let us know! We'll close this issue if you found what you need. thanks!

**Comment by peastman** [comment]
I'd like to add a vote for this issue, with one clarification: it should only scroll if the output is already scrolled down to the bottom.

I often run calculations the produce output over some extended time, maybe printing another line every ten seconds or so.  I want to sit and watch the output as it appears.  But once it collapses all the output into a scrollable text area, it stops being visible.  I can scroll the output down to the bottom, but when the next line appears, it's always hidden off the bottom.  So I have to sit there with the mouse hovering over it, ready to scroll it down by hand every ten seconds just so I can see the next line that's just appeared.  This gets really irritating!

Of course, I might also want to scroll back through the output and look over it while the code is still running.  If it kept jumping back to the bottom, that would also be irritating.  So this is the behavior I suggest.  If the cell's output is already scrolled to the bottom and new output appears, it should automatically scroll to make that new output visible.  If it's scrolled to anywhere other than the bottom, it should remain where it is when new output appears.

And to be clear, I'm talking about scrolling the output of a particular cell.  I'm not talking about scrolling the whole window.

**Comment by Shrey-S** [comment]
Yeah, I'm looking for a solution to this as well.

**Comment by disimone** [comment]
I would also really appreciate such a feature. The collapsed output cell should stay at the point where I leave the scrollbar. Unless I scroll to the very bottom, then it should be implicitely assumed that I want it to follow the output (behavior like the -f option in tail, for example)

**Comment by jozjan** [comment]
You can use the [scroll_down](https://github.com/ipython-contrib/jupyter_contrib_nbextensions/tree/master/src/jupyter_contrib_nbextensions/nbextensions/scroll_down) extenstion from [jupyter_contrib_nbextensions](https://github.com/ipython-contrib/jupyter_contrib_nbextensions) to get automatic output cell scroll mechanism.
![image](https://user-images.githubusercontent.com/185685/49607613-cdebe680-f996-11e8-94e6-d21f4ca2efcb.png)

**Comment by drbenvincent** [comment]
It might be really nice if you could make cell scrolling for persistent. Cell scrolling seems to reset to False whenever you restart the kernel.

**Comment by D3r3kPark** [comment]
Based on the code here:

https://github.com/ipython-contrib/jupyter_contrib_nbextensions/blob/master/src/jupyter_contrib_nbextensions/nbextensions/scroll_down/config.yaml

It looks like you can just change the default flag to true from false?

**Comment by tkondrashov** [comment]
surprised there isn't more support for this! would love this feature

**Comment by Linusnie** [comment]
I would really like this feature too! Constantly having to scroll down when printing the progress of an algorithm gets tedious quickly

**Comment by ljburtz** [comment]
bump for this feature to be the default!

**Comment by kaankork** [comment]
+1 for such feature

**Comment by generic-github-user** [comment]
I often use Jupyter for machine learning experiments and a build-in auto-scrolling feature would be very useful in cases where only text output is available or it is preferred for whatever reason (e.g., TensorFlow's progress bars). I can get more space by opening a new output panel but it eventually has the same issue.

**Comment by ArEnSc** [comment]
whoa, I last talked about this 5 years ago, I am glad an extension was made!

**Comment by generic-github-user** [comment]
Yep haha, sorry to anyone who I notified by accident ... though there is an extension I think it would make sense if there was an option for this included in Jupyter, even just as a line in the config/settings file.

**Comment by danizil** [comment]
still nothing?

**Comment by tommedema** [comment]
Unfortunately that extension doesn't work with Jupyter Lab

**Comment by chouxxxx** [comment]
Hi all,

I am a bit confused because according to your comments here, the autoscrolling of outputs have never been a default jupyter feature.

When i first installed anaconda and used jupyter on my personal computer 2 years ago i did have that nice autoscrolling feature. It was wonderful really, but mostly i don't recall having installed any extension, i don't even know how to install any.

Recently i installed anaconda and use jupyter on my new work's workstation and the autoscrolling isn't implemented and i feel terrible. The slowly printing outputs are not followed by the screen anymore, using ctrl+enter stays at the top of the code cell without showing the output, and even shift+enter just jumps me to show only the first line of the output at the bottom of my screen instead of either showing it at the top of the screen or showing the last line of the output at the bottom of my screen (i don't recall exactly how it used to work but it was one of these two things and it made sense).

Is that extension really the only solution or was the autoscrolling added to the default jupyter installation at some point and i just don't find how to enable it anymore ?

Have a nice day. :)
